### PR TITLE
feature: support rent as percent income

### PR DESCRIPTION
**Pull request by emilyjablonski**
See #1148 for more information
If a unit has `monthlyRentAsPercentOfIncome` field defined, it will show rent as `X% of Income`. This is required for the Nova listing.

Otherwise it will use the `monthlyRent` field as it already is. Below are two screenshots with the change of each approach.
(1) As percent of income
Screen Shot 2021-04-30 at 2 25 55 PM
(2) As monthly rent
Screen Shot 2021-04-30 at 2 26 06 PM

Update: I also pulled in the changes from housingbayarea to toggle the application section on and off based on application methods and show the referral component instead :) Since the strings are Alameda specific I did not include those strings here.

After looking into how to accomplish the new "referral listing" type (no application sections will show), I decided for now to not use a new application method, since those currently exist in the data model as an array that could include multiple methods. The existence of the referral type implies that no internal application methods exist at all. It cannot exist in tandem with other methods, so the toggle logic is that if there are not any methods, referral will show. Open to feedback!

### CODE REVIEW

**Review by software-project — CHANGES_REQUESTED** [review]
Nice job, only thing is that phone numer that we might want to get from listing.

File: sites/public/pages/listing.tsx [hunk 1]
```diff
@@ -307,10 +308,18 @@ export default class extends Component<ListingProps> {
...
-                    internalFormRoute="/applications/start/choose-language"
-                  />
+                  {listing.applicationMethods.length > 0 ? (
+                    <ApplicationSection
+                      listing={listing}
+                      internalFormRoute="/applications/start/choose-language"
+                    />
+                  ) : (
+                    <ReferralApplication
+                      phoneNumber={"211"}
```
[the remark this answers is not in the document]
Comment (emilyjablonski) [reply sites/public/pages/listing.tsx:318]: It's just passed in like a string would be, the same way the string is specific to the county. I can leave it empty in core if you'd like?

[the remark this answers is not in the document]
Comment (software-project) [reply sites/public/pages/listing.tsx:318]: In that case I would pass it as a translation and override it per county basis within translations overrides, so we keep numer od changes in the code to the minimum.

[the remark this answers is not in the document]
Comment (emilyjablonski) [reply sites/public/pages/listing.tsx]: Sounds good! I made it a translatable string and it'll live in per county overrides 👍

**Review by software-project — APPROVED** [review]
LGTM 👍

**Pull request merged**